client/client_test.go

```diff
@@ -918,6 +930,77 @@ func testListTarget(t *testing.T, rootType string) {
 	assert.Equal(t, currentTarget, newCurrentTarget, "current target does not match")
 }
 
+func testListTargetWithDelegates(t *testing.T, rootType string) {
```

Comment:
Actually, apologies - I'm wondering if we need to test both the root being ECDSA and RSA for all these cases.  Maybe this would be fine just being its own test case with a capitalized T?  Again, only if I haven't just blindly missed the function call